File: findthequeen.js
// set up and relate to canvas
var my_canvas = document.getElementById('canvas');
var context = my_canvas.getContext('2d');
my_canvas.addEventListener("mousedown", doMouseClick, false);

var suits = ["Clubs", "Hearts", "Spades", "Diamonds"];
var values = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King", "Ace"];

var deck = [];

function Card(value, suit) {
    this.value = value;
    this.suit = suit;
    this.name = this.value + " of " + this.suit;
	this.isRed = function() {
		if (this.suit === "Diamonds" || this.suit === "Hearts") {
			return true;
		} else {
			return false;
		}
	};
}

function Player(number, name) {
	this.number = number;
	this.name = name;
	this.points = 0;
}

var updateScores = function() {
	context.fillStyle = 'white';
	context.fillRect(90, 480, 410, 55);
	context.fillStyle = 'black';
	context.fillText(player1.points, 120, 510);
	context.fillText(player2.points, 420, 510);
};

var currentPlayer = function(player) {
	if (player.name === player1.name) {
		context.strokeStyle = 'red';
		context.strokeRect(90, 540, 80, 40);
		context.strokeStyle = 'white';
		context.strokeRect(390, 540, 80, 40);
	} else {
		context.strokeStyle = 'red';
		context.strokeRect(390, 540, 80, 40);
		context.strokeStyle = 'white';
		context.strokeRect(90, 540, 80, 40);
	}
};

var player1 = new Player(1, prompt("Player 1, enter your name:"));
var player2 = new Player(2, prompt("Player 2, enter your name:"));

context.font = "20px Garamond";

context.fillText(player1.name, 100, 560);
context.fillText(player2.name, 400, 560);
updateScores();

// create the deck
for (var suitcounter = 0; suitcounter < 4; suitcounter++) {
    for (var valuecounter = 0; valuecounter < 13; valuecounter++){
        var newCard = new Card(values[valuecounter], suits[suitcounter]);
        deck.push(newCard);
    }
}

// shuffle the cards
var shuffle = function (o) {
	for(var j, x, i = o.length; i; j = Math.floor(Math.random() * i), x = o[--i], o[i] = o[j], o[j] = x);
    return o;
};

shuffle(deck);

/*
// test to make sure all cards are in deck

context.fillStyle = 'black';
context.font = "8px Garamond";
var yPlace = 10;
for (var i = 0; i < deck.length; i++) {
	context.fillText(deck[i].name, 40, yPlace);
	yPlace += 10;
}
*/

// choose a random card from the deck, and return its array index
var usedCards = [];

var drawCard = function() {
	if (deck.length < 1) {
		context.fillStyle = "black";
		context.fillText("Out of Cards! Game Over!", 300, 300);
	}
	var card = deck.pop();
	usedCards.push(card);
	context.fillStyle = '#FFFFAA';
	context.fillRect(200, 50, 125, 200);
	if (card.isRed()) {
		context.fillStyle = 'red';
	} else {
		context.fillStyle = 'black';
	}
	context.fillText(card.value, 215, 75);
	context.fillText(card.suit, 220, 150);
	if (card.value === "10") {
		context.fillText(card.value, 300, 230);
	} else {
		context.fillText(card.value[0], 300, 230);
	}
	return card;
};

var play = function(player, opponent, card) {
	currentPlayer(player);
	player.points++;
	if (card.suit === "Hearts") {
		player.points++;
		updateScores();
	} else {
		context.fillStyle = 'black';
		context.fillText("Opponent's turn", 50, 50);
	}
};

// 'pick' button
context.fillStyle = 'green';
context.fillRect(450, 20, 150, 40);

context.fillStyle = 'white';
context.fillText("Pick a Card", 480, 45);

// Click "Pick a Card!" button and draw the chosen card on the canvas
function doMouseClick(event) {
	var canvasX = event.pageX;
	var canvasY = event.pageY;
	if (canvasX >= 450 && canvasX <= 600 && canvasY >= 20 && canvasY <= 80) {
		var drawnCard = drawCard();
		while (drawnCard.isRed()) {
			play(player1, player2, drawnCard);
			drawnCard = drawCard();
		}
	}
};

File: findthequeen.py
import random
import time

# create Players
class Player(object):
	def __init__(self, num, name):
		self.num = num
		self.name = name
		self.points = 0

# make the deck of cards
suits = ["Clubs", "Hearts", "Diamonds", "Spades"]
values = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King", "Ace"]

class Card(object):
	def __init__(self, value, suit):
		self.value = value
		self.suit = suit
		if self.suit == "Clubs" or self.suit == "Spades":
			self.color = "black"
		else:
			self.color = "red"
		if self.value == "Jack" or self.value == "Queen" or self.value == "King" or self.value == "Ace":
			self.isFace = True
		else:
			self.isFace = False
		self.name = str(self.value + " of " + self.suit)

deck = []
discardPile = []

for suit in suits:
	for value in values:
		deck.append(Card(value, suit))

# Start the game
print "----Find the Queen!----"
print "Welcome! Player 1, please enter your name:"
name = raw_input()
player1 = Player(1, name)
print "Player 2, please enter your name:"
name = raw_input()
player2 = Player(2, name)
		
random.shuffle(deck)
"""
def cardManager(pickedCard):
	discardPile = []
	discardPile.append(deck.pop(pickedCard))
	return discardPile
"""
"""
The game begins...
Pick a red card, and you get a point. If that red card is a face card, you get another point!
If that red face card happens to be the King of Hearts, though, you lose all points!
If it's the Queen of Diamonds, though, you get get an EXTRA 5 points!
If a red Ace is pulled, the other team gets a point too.
Keep picking card until you get a black card.
If that black card was the Ace of Spades, your opponent scores 5 points and starts their turn.
If it was the 2 of Clubs, all the cards get shuffled back into the deck and you continue playing!
"""
usedCards = []
def playGame(player, opponent):
	time.sleep(1)
	print "%s's turn! ------------------------------ Score: %s: %d   %s: %d" % (player.name, player.name, player.points, opponent.name, opponent.points)
	nextCard = raw_input("Type 'pick' to choose your next card! > ")
	while nextCard != "pick":
		nextCard = raw_input("Whoops! Try again: type 'pick' to choose your next card! > ")
	pickedCard = deck.pop()
	usedCards.append(pickedCard)
	print "You picked the %s." % pickedCard.name

	while pickedCard.color == "red":
		player.points += 1
		time.sleep(2)
		print "Red Card! %s got a point! You have %d points!" % (player.name, player.points)
		if pickedCard.isFace:
			player.points +=1
			time.sleep(2)
			print "Face card! You got a point! You have %d points!" % player.points
			if pickedCard.name == "King of Hearts":
				player.points = 0
				time.sleep(2)
				print "Bummer...the King of Hearts. Back to 0 points."
			if pickedCard.name == "Queen of Diamonds":
				player.points += 5
				time.sleep(2)
				print "Sweet! The Queen of Diamonds! Plus 5 points! You have %d points!" % player.points
			if pickedCard.value == "Ace":
				opponent.points += 1
				time.sleep(2)
				print "That Ace was good for you, AND the other team! They just got a point and have %d points!" % opponent.points
		nextCard = raw_input("Type 'pick' to choose your next card! > ")
		while nextCard != "pick":
			nextCard = raw_input("Whoops! Try again: type 'pick' to choose your next card! > ")
		try:
			pickedCard = deck.pop()
			usedCards.append(pickedCard)
		except:
			print "You've run out of cards!"
		print "You picked the %s." % pickedCard.name
	if pickedCard.name == "Ace of Spades":
		opponent.points += 5
		time.sleep(2)
		print "Bummer! The Ace of Spades! The other team just got 5 points...They have a total of %d points." % opponent.points

	if pickedCard.name == "2 of Clubs":
		time.sleep(2)
		random.shuffle(deck)
		print "The 2 of Clubs! Time to shuffle the deck again..."
	return usedCards

while len(usedCards) < 52:
	playGame(player1, player2)
	print len(usedCards)
	playGame(player2, player1)
	print len(usedCards)
	
print "You went through the whole deck! Game over! %s had %d points. %s had %d points." % (player1.name, player1.points, player2.name, player2.points)
if player1.points > player2.points:
	time.sleep(2)
	print "%s wins!" % player1.name
elif player2.points > player1.points:
	time.sleep(2)
	print "%s wins!" % player2.name
else:
	print "Tie game!"